spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerMethodTests.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.SubscribableChannel;
+
+/**
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+public class StreamListenerMethodTests {
+
+
+	@Test
+	public void testMethodWithInputAsMethodAndParameter() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithInputAsMethodAndParameter.class, "--server.port=0");
+			fail("Exception expected on using Input element as method and parameter");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).contains("@Input annotation can not be used in multiple method parameters");
+		}
+	}
+
+	@Test
+	public void testMethodWithOutputAsMethodAndParameter() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithOutputAsMethodAndParameter.class, "--server.port=0");
+			fail("Exception expected on using Input element as method and parameter");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("Cannot set both Output (@Output/@SendTo) method annotation value ");
+		}
+	}
+
+	@Test
+	public void testMethodWithoutInput() throws Exception {
+		try {
+			SpringApplication.run(TestMethodWithoutInput.class, "--server.port=0");
+			fail("Exception expected when inbound target is not set");
+		}
+		catch (BeanCreationException e) {
+			assertThat(e.getCause().getMessage()).startsWith("Either StreamListener or a method parameter should be set with ");
```

Comment:
this as well was added with `startsWith`. The actual message is: `Either StreamListener or a method parameter should be set with an inbound element`. Can add your suggestion there.